File: 513.找树左下角的值.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int findBottomLeftValue(TreeNode* root) {
        // 从右往左层次遍历
        queue<TreeNode*> q;
        q.push(root);
        TreeNode* t;
        while(!q.empty()){
            t = q.front();
            q.pop();
            if(t->right)
                q.push(t->right);
            if(t->left)
                q.push(t->left);
        }
        return t->val;
    }
};

File: 二叉树/513.找树左下角的值.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
 var findBottomLeftValue = function(root) {
    let queue = [root];
    let res;
    while (queue.length) {
        let size = queue.length;
        res = queue[0].val;
        while (size--) {
            let node = queue.shift();
            if (node.left !== null)
                queue.push(node.left);
            if (node.right !== null)
                queue.push(node.right);
        }
    }
    return res;
};